public/editor/scripts/project/project.js

```diff
@@ -91,6 +91,18 @@ define(function(require) {
     Metadata.removeFile(getRoot(), stripRoot(path), callback);
   }
 
+  // Sets a flag on the project root that indicates whether we need to update
+  // the published version of this project or not
+  function publishNeedsUpdate(value, callback) {
+    Metadata.setPublishNeedsUpdate(getRoot(), value, callback);
+  }
+
+  // Gets the flag from the project root that indicates whether we need to
+  // update the published version of this project or not
+  function doesPublishNeedUpdate(callback) {
```

Comment:
I would call this `getPublishNeedsUpdate` (also note the `s` on `Needs` that you missed).